### PR TITLE
Added jwt token/profile into redux && switched okta to auth0

### PR DESCRIPTION
**Pull request by JordonP123**
**Description:** 

Set up Auth0 so a user can be authenticated.  Added jwt/profile into redux, so we can hit endpoints and know who we are at any point.  

- What work was done?
  - Disassembled Okta and assembled Auth0
  - Added private routes to each component that needs to be private
  - Fixed API calls to correlate with auth0
  - Added profile/token reducers and actions 
  - Set profile into redux once token is verified
  
- What feature / user story is it for?
  - [Jira](https://bloomtechlabs.atlassian.net/jira/software/c/projects/BL/boards/7?modal=detail&selectedIssue=BL-1077)

- [LOOM](https://www.loom.com/share/f3f8dbee2d2047f581c1ded0389ae548)

### CODE REVIEW

File: src/components/common/LoadingComponent.js [hunk 1]
```diff
@@ -7,6 +7,8 @@ function LoadingComponent(props) {
   return <div>{message}</div>;
 }
 
+//this is for no reason
```
Comment (Kobramall) [inline src/components/common/LoadingComponent.js:10]: probably should have deleted this haha

File: src/state/reducers/profileReducer.js [hunk 2]
```diff
@@ -0,0 +1,16 @@
+import { setProfile } from '../actions';
+
+const intialState = {
```
Comment (Kobramall) [inline src/state/reducers/profileReducer.js:3]: nothing seems wrong from what I can see

File: src/state/actions/index.js [hunk 3]
```diff
@@ -3,5 +3,13 @@ import * as contactPreferences from './contactPreferencesActions';
 import * as referral from './referralsActions';
 import * as document from './documentUploadActions';
 import * as setAuthToken from './authTokenAction';
+import * as setProfile from './profileAction';
 
-export { client, contactPreferences, referral, document, setAuthToken };
+export {
```
Comment (ramib5734) [inline src/state/actions/index.js:8]: Love the organization of this code! Reads much cleaner then when it's in one line.

File: src/components/common/PrivateRoute.js [hunk 4]
```diff
@@ -3,22 +3,28 @@ import { Route } from 'react-router-dom';
 import { withAuthenticationRequired } from '@auth0/auth0-react';
 import { connect } from 'react-redux';
 import * as actions from '../../state/actions/authTokenAction';
+import * as actions1 from '../../state/actions/profileAction';
```
Comment (ramib5734) [inline src/components/common/PrivateRoute.js:6]: maybe this could be renamed from actions1 to something else more related to the type of action?

**Review by ramib5734 — COMMENTED** [review]
Some general comments about the code. Looks good though!

**Pull request merged**